test_utils/utils.go

```diff
@@ -142,7 +142,7 @@ func CreateContainer(netconf string, k8sName string) (container_id, netnspath st
 	//TODO pass in the env properly
 	var k8s_env = ""
 	if k8sName != "" {
-		k8s_env = fmt.Sprintf("CNI_ARGS=\"K8S_POD_NAME=%s;K8S_POD_NAMESPACE=test;K8S_POD_INFRA_CONTAINER_ID=whatever\"", k8sName)
+		k8s_env = fmt.Sprintf("CNI_ARGS=\"K8S_POD_NAME=%s;K8S_POD_NAMESPACE=test;K8S_POD_INFRA_CONTAINER_ID=whatever;IP=%s\"", k8sName, ip)
```

Comment:
Rather than always append this, could you have a separate IF statement which adds it if `ip` is passed?

Otherwise we're changing all the other tests to get passed `IP=`